site/src/components/__tests__/topbar.js

```diff
@@ -19,8 +20,12 @@ describe("TopBar", () => {
   }
 
   it("renders correctly", () => {
-    const tree = renderer.create(<TopBar />).toJSON()
+    let tree = renderer.create(<TopBar />).toJSON()
     expect(tree).toMatchSnapshot()
+    process.env.GATSBY_SITE_BANNER_COLOUR = "secondary"
+    tree = renderer.create(<TopBar />).toJSON()
+    expect(tree).toMatchSnapshot()
+    process.env.GATSBY_SITE_BANNER_COLOUR = "primary"
```

Comment:
Less brittle to record the value or the env var just before you set it, then set it back to the recorded value afterwards. Then you make no assumptions about the original value.